From e6bd2919e8b99c82b068426ea6572c0021b1f155 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Fri, 25 Jun 2021 15:18:09 +0200
Subject: [PATCH] Link against QtWidgets

We have a non-trivial dep on QtWidgets (QApplication, QFileDialog)

Make this explicit
---
 CMakeLists.txt     | 2 +-
 src/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4b8d786..519e405 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,7 @@ ecm_setup_version(${PROJECT_VERSION}
     VARIABLE_PREFIX PLASMAPHONEBOOK
     VERSION_HEADER ${CMAKE_CURRENT_BINARY_DIR}/version.h
 )
-find_package(Qt5 ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Core Quick Test Gui Svg QuickControls2)
+find_package(Qt5 ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Core Quick Test Gui Svg QuickControls2 Widgets)
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS CoreAddons Kirigami2 People Contacts)
 find_package(KF5PeopleVCard REQUIRED)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6f97349..65ee4d5 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -10,7 +10,7 @@ add_executable(plasma-phonebook
 target_include_directories(plasma-phonebook PRIVATE ${CMAKE_BINARY_DIR})
 
 target_link_libraries(plasma-phonebook
-        Qt::Core Qt::Qml Qt::Quick Qt::Svg
+        Qt::Core Qt::Qml Qt::Quick Qt::Svg Qt::Widgets
         KF5::Contacts KF5::People KF5::I18n KF5::CoreAddons
 )
 
-- 
2.32.0

